github-pr-background.js

```diff
@@ -53,19 +53,21 @@ async function createAttachTab(msg) {
     var attValue = sanitizeForTemplate(msg.prURL);
     var descValue = sanitizeForTemplate("pr " + msg.prNum + ": " + msg.prTitle);
 
-    // We need to add values for this specific attachment, so we
-    // build the template with local values and then execute the
-    // code in the tab.
+    // We need to add values for this specific attachment, so we build the
+    // template with local values and then execute the code in the tab.
     var attachScript = `
-        // switch from upload-file to paste-text attachment
-        if (document.querySelector(".attachment_text_field.bz_tui_hidden")) {
-            document.getElementById("attachment_data_controller").click();
-        }
-
-        let att = document.getElementById("attach_text");
-        let desc = document.getElementById("description");
+        // Add PR link
+        let att = document.getElementById("att-textarea");
         att.value = "${attValue}";
-        desc.value = "${descValue}";
+        // Trigger an input event so the textarea input handling kicks off.
+        var input_event = new Event('input', {
+            bubbles: true,
+            cancelable: true
+        });
+        att.dispatchEvent(input_event);
```

Comment:
Is this a good way to create and dispatch an event? Are there better ways? Does it have appropriate properties?